src/ray/core_worker/transport/thread_pool.h

```diff
@@ -37,10 +39,12 @@ class BoundedExecutor {
     return max_concurrency_in_default_group > 1 || has_other_concurrency_groups;
   }
 
-  explicit BoundedExecutor(int max_concurrency);
+  explicit BoundedExecutor(
+      int max_concurrency,
+      std::function<std::function<void()>()> initialize_thread_callback = nullptr);
 
   /// Posts work to the pool
-  void Post(std::function<void()> fn) { boost::asio::post(*pool_, std::move(fn)); }
+  void Post(std::function<void()> fn);
```

Comment:
please improve the header comment:

- Is this async or does it block if all threads are busy?
- Is there any guarantee about the execution order on the set of threads? (round robin, for example). IMO we shouldn't rely on any specific behavior.

Reply:
Updated: https://github.com/ray-project/ray/pull/52575/commits/3a641b4fe2d5d0bf87928555723083b062c241fb

> Is this async or does it block if all threads are busy?

It's non-blocking.

https://live.boost.org/doc/libs/1_83_0/doc/html/boost_asio/reference/io_context/post.html